src/hb-ot-math-table.hh

```diff
@@ -344,26 +344,32 @@ struct MathKern
     const MathValueRecord* kernValue = mathValueRecordsZ.arrayZ + heightCount;
     int sign = font->y_scale < 0 ? -1 : +1;
 
-    /* The description of the MathKern table is a ambiguous, but interpreting
-     * "between the two heights found at those indexes" for 0 < i < len as
-     *
-     *   correctionHeight[i-1] < correction_height <= correctionHeight[i]
-     *
-     * makes the result consistent with the limit cases and we can just use the
-     * binary search algorithm of std::upper_bound:
```

Comment:
With your changes, isn't the loop below actually exactly doing the same as `std::upper_bound` (see https://en.cppreference.com/w/cpp/algorithm/upper_bound#Possible_implementation)?

If so, I think it's not necessary to add a comment with the explanation of the algorithm below. Instead, I'd keep the reference to `std::upper_bound` and explain that this matches the spec, including for the boundary cases.

Reply:
You are right. That's just what `std::upper_bound()` does.
I have changed the comment as you suggested.